api/v1alpha3/haproxyloadbalancer_types.go

```diff
@@ -37,6 +38,86 @@ type HAProxyLoadBalancerSpec struct {
 	// deployed VM.
 	// +optional
 	User *SSHUser `json:"user,omitempty"`
+
+	// ExtraServices is additional services that will proxied to.
+	// +optional
+	ExtraServices []ExtraService `json:"extraServices,omitempty"`
+}
+
+// ExtraServices defines additional services that will be proxied by HAProxyLoadBalancer.
+type ExtraService struct {
+	// The name of this service
+	Name string `json:"name"`
+
+	// The list of ports that are exposed by this service.
+	Ports []ExtraServicePort `json:"ports"`
+
+	// Label selector for backend servers. Machines matching
+	// the labels will be selected as targets for the backend servers
+	// +optional
```

Comment:
This should be required, otherwise you end up with empty backends.

Reply:
By default if the labels are empty it would pick up all the machines. Is this behavior undesirable?